### DISCUSSION TITLE
Is this project still being maintained?

### DISCUSSION THREAD

**Discussion by Stef16Robbe** [cdk8s-team/cdk8s#2052] [Q&A] [answered]
Hi,

First of all, just wanted to say I'm really impressed with `cdk8s` and I'm looking forward to using it.

I was just wondering if the project is still being maintained before I commit a bunch of time to learning/integrating the tool. I'm asking because the last non-automated commit to the Master branch was in [October of last year](https://github.com/cdk8s-team/cdk8s/commits/master/?after=09b39042b6bfa7ae2d73be403eac1e688df6e220+209). And Issues/PR's seem to run stale as well. The same seeming to be the case for the other repo's.

Thanks!

**Comment by RonaldGalea** [comment]
Would like to hear an update as well. Project and idea sound great, would love to be able to use something like this.

**Comment by plaisted** [comment]
I am using the project for managing kubernetes resources and it works great for my use cases but the lack of activity / bug fixes is concerning. I hope this keeps being maintained as it creates a nice bridge between infrastructure managed using CDK and kubernetes resources (sharing config objects, cdk generated values (load balancers etc)).

**Comment by plaisted** [comment]
It's completely valid to ask what the status of an open source project is, especially one developed at some point by a large company and accepted to CNCF as Incubating.

AWS makes plenty of money from us including spend on EKS / EC2 driven by this library. I am asking our TAM about the project to get an answer. My comment here was to add feedback that the library is being used and brings real value.

**Comment by RonaldGalea** [comment]
For others interested in not writing YAML by hand anymore - in case it turns out CDK8s is not maintained anymore, would using the [K8s client libraries directly to generate YAML](https://github.com/learnk8s/templating-kubernetes/tree/master/python) be a viable option? It would essentially be more bare-bones tooling, but on the flip side, very stable. Thoughts? Has anyone already tried and can share the experience?

**Comment by rassie** [comment]
> For others interested in not writing YAML by hand anymore - in case it turns out CDK8s is not maintained anymore, would using the [K8s client libraries directly to generate YAML](https://github.com/learnk8s/templating-kubernetes/tree/master/python) be a viable option? It would essentially be more bare-bones tooling, but on the flip side, very stable. Thoughts? Has anyone already tried and can share the experience?

FWIW, you can do the same with just `cdk8s-core`. You'd import `k8s@1.XX` and then work with `Kube*` classes. This part of `cdk8s` should be good for the foreseeable future since it doesn't really depend on Kubernetes version or anything else.

**Comment by MatthewCane** [comment]
@plaisted did you get a response from your account manager? I am going to ask the same question to mine but before I do I thought I would see if you've got any updates from yours.

**Comment by iliapolo** [comment]
We’re currently forming our plans for the remainder of the year and will inform the community within the next few weeks.

**Comment by MatthewCane** [comment]
Thank you for the update

**Comment by nicolai0** [comment]
@iliapolo any news on this? All of my k8s configuration is in Cdk8s, it would be sad to see it deprecated and have to got back to Yaml :(

**Comment by Stef16Robbe** [comment]
@nicolai0 My guess is that they will continue this project, back in April they presented CDK8S at [AWS Summit Amsterdam](https://aws.amazon.com/events/summits/emea/amsterdam/agenda/?emea-event-agenda-card.sort-by=item.additionalFields.title&emea-event-agenda-card.sort-order=asc&awsf.emea-event-agenda-level=*all&awsf.emea-event-agenda-role=*all&awsf.emea-event-agenda-category=*all&awsf.emea-event-agenda-aws-industry=*all&emea-event-agenda-card.q=DOP301&emea-event-agenda-card.q_operator=AND#session).

I wasn't there personally though.

**Comment by iliapolo** [comment]
Hey Folks. As mentioned earlier, we are forming our plans for the remainder of the year. The process takes some time but we are in the final stages. We will update this issue once we have those plans in place. 

Appreciate your patience.

**Comment by saiyush** [comment]
Hi all - we understand the community’s recent concerns around CDK8s and we want to assure everyone that they’ve been heard. Currently, AWS is still actively maintaining CDK8s, taking care of critical issues that arise. However, we are working behind the scenes to define a clear roadmap for the project’s future, including how it will be managed moving forward. I want to assure everyone that all the valuable feedback from the community is being considered. While this may not be the definitive answer everyone was hoping for, we plan to provide a more concrete update by Q4 2024. In the meantime, we highly encourage everyone to keep providing their feedback.

**Reply by nilroy** [reply]
@saiyush Any update on this?

**Comment by XLordalX** [comment]
Any updates?

**Comment by bdoyle0182** [comment]
This is an extremely valuable project for us and getting to a fully cohesive CDK infrastructure story for both public cloud configuration (AWS CDK) and cloud native configuration (CDK8s), but actually choosing to adopt it for production k8s configuration widely at a platform level depends on it still being maintained and invested in. CDK as a model / framework for IaC is the future and the value this provides with giving developers a singular ecosystem for building infrastructure within the context of that framework is invaluable. It's now midway through Q4, are any updates coming soon @saiyush

**Comment by nraooptum** [comment]
We're seriously exploring cdk8s as a possibility to replace our traditional Terraform-based IaC in my organization (a multi-billion-dollar org in a larger Fortune 10 enterprise). Pairing cdk8s with Azure Service Operator, ArgoCD, and ORAS (OCI artifacts) is proving to be an invaluable combination. Looking forward to updates @saiyush @iliapolo

**Reply by shinebayar-g** [reply]
If you're not AWS customer, you have no say here. /s

**Reply by nraooptum** [reply]
Ha, we are on all the major cloud platforms. Except AWS. Kidding, we have a big footprint on AWS as well.

**Reply by shinebayar-g** [reply]
You should better talk to your account manager. They should have a better handle on this.

**Reply by nilroy** [reply]
@nraooptum Have you heard anything from your AWS account manager? Recently they added the [Roadmap](https://github.com/cdk8s-team/cdk8s/blob/master/ROADMAP.md#roadmap) file in the repo pointing to this discussion.

**Comment by shinebayar-g** [comment]
Hey all, I hope it's okay to post here.

I built a lightweight alternative to cdk8s. Feel free to checkout here https://github.com/konfjs/k8skonf.

Project is heavily inspired by Pulumi/CDKTF/CDK8S and API is very similar. However, it's not built on Constructs or JSII. So it's Typescript only. But the API is clean as a result. 

Since general API is more or less similar to cdk8s, migration path should be straightforward. It's pure library and doesn't require additional CLI. So it should be fully embeddable in any project.

**Comment by jareks** [comment]
I just wanted to say that this is a fantastic tool and it would be horrible to lose it.

**Reply by shinebayar-g** [reply]
Alternatives are already there :) I'm not currently missing anything from cdk8s yet.

**Reply by fernandonogueira** [reply]
@shinebayar-g Which alternative are you using?

**Reply by shinebayar-g** [reply]
Mentioned just above https://github.com/konfjs/k8skonf.

**Comment by burner1024** [comment]
So who is discussing what with whom? It's going to be a full year soon.

With alternatives rising, it's becoming use it or lose it situation for AWS.

**Comment by Stef16Robbe** [comment]
Found this in the [CDK slack](https://cdk-dev.slack.com/archives/C019M3CPSV6/p1736538209812699?thread_ts=1731957505.308989&cid=C019M3CPSV6), posted January 10th:

> All - We’re currently working on the CDK plan for 2025. We’re thinking about including support for CDK8s and L2 for Bedrock in this plan. We don’t have things finalized yet, but we’ll share our updated plan, including information about these items by next month.

**Comment by mildronize** [comment]
I've found the alternative TypeScript library: https://github.com/tommy351/kosko

**Comment by mukundjalan** [comment]
@saiyush is there any update on this? If not, do we have any alternative option for python?

**Comment by saiyush** [comment, marked as answer]
We appreciate your patience while we evaluated the future support of CDK8s. We're delighted to confirm that the CDK team will maintain their support for CDK8s moving forward. You can find additional information in our latest roadmap documentation https://github.com/aws/aws-cdk/blob/main/ROADMAP.md#objective-3---extensible-infrastructure:~:text=patterns%20and%20practices.-,Work%20in%20progress,-Continued%20support%20for

**Comment by bdoyle0182** [comment]
Glad to hear that it will be maintained and the AWS vision is to invest in CDK as holistic model for IaC and also see better integration with partners that are also investing in cdk's for their products. Will this include investment in new features / what would be considered r&d? or is the state of project strictly maintenance and resolving issues? Many of the open github issues are major feature requests and vision for how the project can improve rather than just bug resolution

**Comment by rassie** [comment]
@saiyush So what's the plan now? Over 18 months after the announcement we are nowhere near sustained maintenance of CDK8S. While I'm still confident that `cdk8s-core` is mostly enough for most tasks and can be considered stable, `cdk8s-plus` has yet to be released for Kubernetes 1.35, released last December. Still, issues both by humans and bots alike are piling up with no action in sight. 

I think there is a lot of appetite to see CDK8S develop, but as things look right now, the project needs to expand ownership and maintenance onto willing community members outside of AWS.
